Update filters to take gen energy for 8 GeV Update

JYoo001 opened the pull request:
I am updating _ldmx-sw_, here are the details.

### What are the issues that this addresses?
Resolves #1696 
Also a solution to #1702 


## Check List
- [x] I successfully compiled _ldmx-sw_ with my developments.
- [x] I read, understood and follow the [coding rules](https://ldmx-software.github.io/developing/coding-rules.html).
<!-- If these coding rules are confusing or you need help, still open the PR as a _draft_ and we can help you! -->
- [ ] I ran my developments and the following shows that they are successful.
<!-- put plots or some other proof that your developments work and do the intended function -->

tvami commented:
```
  ---- LDMXSW: Loading configuration --------
  Traceback (most recent call last):
    File "config.py", line 9, in <module>
      mySim = ecal.photo_nuclear(det,gen.single_8gev_e_upstream_tagger())
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/runner/work/ldmx-sw/ldmx-sw/install/python/LDMX/Biasing/ecal.py", line 73, in photo_nuclear
      filters.TaggerVetoFilter(generator = generator),
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/runner/work/ldmx-sw/ldmx-sw/install/python/LDMX/Biasing/filters.py", line 197, in __init__
      self.threshold = thresh
                       ^^^^^^
  NameError: name 'thresh' is not defined
  ```
  @JYoo001 did you see this in the CI?

JYoo001 commented:
Yes, I'll go update it right now!

JYoo001 committed:
Filter default updated, changes in ecal, target, eat

JYoo001 commented:
ignore for now, just rebasing

JYoo001 commented:
I've updated ecal.py, eat.py, and target.py to calculate the run parameters to give to filters.py. I've also updated the default values in filters.py, and passed in inputs in some of the other files that relied on the defaults. I've left the dark brem biasing untouched as well.

tvami commented on Biasing/python/filters.py:
I'm wondering if in this file, we should just not have defaults at all. That will force the user to have the values passed wherever these are used, and that kinda ensures that we never end up in this situation of forgetting to overwrite certain settings. Like I think at this point Jihoon did add the explicit values in all occurrences, so why not just remove the defaults?

tomeichlersmith replied:
I second this idea.

tvami commented on Biasing/python/filters.py:
```diff
@@ -29,7 +29,7 @@ class TargetBremFilter(simcfg.UserAction):
     """
 
-    def __init__(self,recoil_max_p = 1500.,brem_min_e = 2500.) :
+    def __init__(self,recoil_max_p = 3000.,brem_min_e = 5000.) :
```
```suggestion
    def __init__(self,recoil_max_p,brem_min_e) :
```

tvami commented on Biasing/python/filters.py:
```diff
@@ -50,7 +50,7 @@ class NonFiducialFilter(simcfg.UserAction):
     """
 
-    def __init__(self,recoil_max_momentum = 1500., abort_fid_event = True) :
+    def __init__(self,recoil_max_momentum = 3000., abort_fid_event = True) :
```
```suggestion
    def __init__(self,recoil_max_momentum, abort_fid_event = True) :
```

tvami commented on Biasing/python/filters.py:
```diff
@@ -113,7 +113,7 @@ class TargetENFilter(simcfg.UserAction) :
     """
 
-    def __init__(self,recoil_thresh = 2500.) :
+    def __init__(self,recoil_thresh = 5000.) :
```
```suggestion
    def __init__(self,recoil_thresh) :
```

tvami commented on Biasing/python/filters.py:
```diff
@@ -188,7 +188,7 @@ class TaggerVetoFilter(simcfg.UserAction):
     """
     
-    def __init__(self,thresh=3800., reject_events_missing_tagger=True) :
+    def __init__(self,thresh=7600., reject_events_missing_tagger=True) :
```
```suggestion
    def __init__(self,thresh, reject_events_missing_tagger=True) :
```

JYoo001 committed:
Update Biasing/python/filters.py

Co-authored-by: Tamas Vami <tamas.almos.vami@cern.ch>

JYoo001 committed:
Update Biasing/python/filters.py

Co-authored-by: Tamas Vami <tamas.almos.vami@cern.ch>

JYoo001 committed:
Update Biasing/python/filters.py

Co-authored-by: Tamas Vami <tamas.almos.vami@cern.ch>

JYoo001 committed:
Update Biasing/python/filters.py

Co-authored-by: Tamas Vami <tamas.almos.vami@cern.ch>
